dashboard/app/models/levels/level.rb

```diff
@@ -350,6 +350,37 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  def log_changes(user=nil)
+    return unless changed?
+
+    log = JSON.parse(audit_log || "[]")
+
+    # gather all field changes; if the properties JSON blob is one of the things
+    # that changed, rather than including just 'properties' in the list, include
+    # all of those attributes within properties that changed.
+    latest_changes = changed.dup
+    if latest_changes.include?('properties') && changed_attributes['properties']
+      latest_changes.delete('properties')
+      changed_attributes['properties'].each do |key, value|
+        latest_changes.push(key) unless properties[key] == value
+      end
+    end
+
+    entry = {
+      changed_at: Time.now,
+      changed: latest_changes
+    }
+    entry[:changed_by] = user.id unless user.nil?
```

Comment:
Using user.email would be my preference. Unfortunately, student accounts don't have (non-blank) email addresses.